libcommon/touch.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2023 - Tillitis AB
+ * SPDX-License-Identifier: GPL-2.0-only
+ */
+
+#include <stdbool.h>
+#include <tkey/led.h>
+#include <tkey/touch.h>
+#include <tkey/qemu_debug.h>
+
+// CPU clock frequenzy in Hz
+#define CPUFREQ 18000000
+
+//#define timerrunning() (*timer_status & (1 << TK1_MMIO_TIMER_STATUS_RUNNING_BIT))
+
+// clang-format off
+static volatile uint32_t *timer =           (volatile uint32_t *)TK1_MMIO_TIMER_TIMER;
+static volatile uint32_t *timer_prescaler = (volatile uint32_t *)TK1_MMIO_TIMER_PRESCALER;
+static volatile uint32_t *timer_status =    (volatile uint32_t *)TK1_MMIO_TIMER_STATUS;
+static volatile uint32_t *timer_ctrl =      (volatile uint32_t *)TK1_MMIO_TIMER_CTRL;
+static volatile uint32_t *touch = (volatile uint32_t *)TK1_MMIO_TOUCH_STATUS;
+// clang-format on
+
+bool touchwait(int color, int timeout)
+{
+	int ledon = 0;
+	int origled = getled();
+	uint32_t time = 0;
+	uint32_t lasttime = 0;
+
+	// Tick once every decisecond
+	*timer_prescaler = CPUFREQ / 10;
+	*timer = timeout * 10; // Seconds
+
+	// Start timer
+	*timer_ctrl = (1 << TK1_MMIO_TIMER_CTRL_START_BIT);
+
+	// Acknowledge any stray touch events before waiting for real
+	// touch
+	*touch = 0;
+
+	// Blink until either the touch sensor has been touched or the
+	// timer hits 0.
+	while (!touched() && *timer_status != 0) {
+		time = *timer;
+		if (time % 2 == 0 && time != lasttime) {
+			lasttime = time;
+			ledon = !ledon;
+			setled(ledon ? color : LED_BLACK);
+		}
+	}
+
+	// Restore LED
+	setled(origled);
+
+	// Do we have a timeout?
+	if (*timer_status == 0) {
+		return false;
+	}
+
+	// Stop timer
+	*timer_ctrl = (1 << TK1_MMIO_TIMER_CTRL_STOP_BIT);
```

Comment:
Shouldn't this be  `*timer_ctrl |= (1 << TK1_MMIO_TIMER_CTRL_STOP_BIT);` so we are not breaking things if for some reason we do get more bits to control in the future?